libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -238,64 +265,122 @@ string IRGenerator::generateInitialAssignment(VariableDeclaration const& _varDec
 	return generator.code();
 }
 
-string IRGenerator::constructorCode(ContractDefinition const& _contract)
+pair<string, map<ContractDefinition const*, string>> IRGenerator::evaluateConstructorArguments(
+	ContractDefinition const& _contract
+)
 {
-	// Initialization of state variables in base-to-derived order.
-	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
-
-	using boost::adaptors::reverse;
-
-	ostringstream out;
-
-	FunctionDefinition const* constructor = _contract.constructor();
-	if (constructor && !constructor->isPayable())
-		out << callValueCheck();
+	map<ContractDefinition const*, string> constructorParams;
+	vector<pair<ContractDefinition const*, std::vector<ASTPointer<Expression>>>> baseConstructorArguments;
```

Comment:
```suggestion
	vector<pair<ContractDefinition const*, std::vector<Expression const*>>> baseConstructorArguments;
```
(we are not owning here)

Reply:
Ah, the other option is to use a pointer to a vector of astpointers of expressions...